docs/components/migrating-to-components.md

```diff
@@ -0,0 +1,90 @@
+# Migrating to Components
+
+After your component has been created you can link it to other components such as flares, status pages, or health.
+
+This page documents how to fully integrate your component in the Agent life cycle.
+
+## Flare
+
+Migrate the code related to your component's domain from `pkg/flare` to your component. After migrating
+the Agent to components, you can delete `pkg/flare`.
+
+### Creating a callback
+
+To add data to a flare, register a provider.
+
+First create a `flare.go` file in your component (the file name is a convention) and create a `func (c *yourComp) fillFlare(fb flarehelpers.FlareBuilder) error` function.
```

Comment:
I think it makes it much easier to work with a feature across multiple component. I was thinking about cases where we want to troubleshoot some behavior, that way we don't "struggle" to find the right piece of code in charge of `flare`, `health` or `status page` in a component.

I was thinking we could expand this to all cross-component feature. Each component would have a `heath.go`, `status_page.go` or `flare.go`. 

Definitely open to suggestion here, but I thought it would be useful.

For the method name, it was to ease searching thought the code. But same as before, open to debate :man_shrugging: .

Reply:
Searching for `flarehelpers.NewProvider` gives all the entry points of the flare. Let's discuss it offline.

Reply:
After discussing: Removing the convention for now, it might not be that useful. We will come back to the idea of convention for cross component feature in the future.